ggml.c

```diff
@@ -193,7 +196,13 @@ typedef void * thread_ret_t;
 #else
 inline static void * ggml_aligned_malloc(size_t size) {
     void * aligned_memory = NULL;
-#ifdef GGML_USE_METAL
+#ifdef GGML_USE_CPU_HBM
+    if (size == 0) {
+        GGML_PRINT("WARNING: Behavior may be unexpected when allocate 0 byte for hbw_posix_memalign!");
+        return NULL;
+    }
```

Comment:
This check should be applied for all branches - it's always UB to allocate 0 bytes

Reply:
done.